hack/lib/etcd.sh

```diff
@@ -16,6 +16,10 @@
 
 # A set of helpers for starting/running etcd for tests
 
+set -o errexit
```

Comment:
I'm not sure we want these, since usually they're set by the script calling them. (or is shellcheck complaining about them missing?)

Reply:
Without this `set -o errexit`, shellcheck complains: 
```
In hack/lib/etcd.sh line 111:
    cd "${KUBE_ROOT}/third_party"
    ^---------------------------^ SC2164: Use 'cd ... || exit' or 'cd ... || return' in case cd fails.
```

@fejta proposed this idea to fix this issue: https://github.com/kubernetes/kubernetes/pull/73252#discussion_r251319014

Reply:
@ixdy it shouldn't hurt?

Reply:
it's kinda inconsistent, though - the other library scripts in `hack/lib` don't set it either (except for `init.sh`, which is how other scripts depend on `etcd.sh` anyway.)

Reply:
we could disable the lint for this file instead, not sure that it's better though

Reply:
@ixdy @BenTheElder otherwise, does this look better : `cd "${KUBE_ROOT}/third_party" || return 1` ?

Reply:
If it's good, `set -o errexit` is not needed anymore.

Reply:
Yeah if you remove these and switch to `cd "${KUBE_ROOT}/third_party" || return 1` I'm happy with it

Reply:
@spiffxp thanks, PR updated, PTAL